app/services/workflow_link_service.rb

```diff
@@ -0,0 +1,28 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
+    TagLink.where(tag_link(options)).first_or_create
+    nil
```

Comment:
Why we force to return `nil` here?

Reply:
The caller only needs to know whether it succeeds or not. No `TagLink` object is required.